### ISSUE TITLE
`datalad.log.LogHelper` uses variable that should be constant

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6730]
See https://github.com/datalad/datalad-extension-template/issues/39

`LogHelper(name=)` could be set different from the default 'datalad'. This feature is not used anywhere. If one would do that, the code would start querying for config items with names that are no supported. I.e. `datalad_osf.log.level`. The ConfigManager does not handle arbitrary names, and the `LoggerHelper` does not make sure that the given name is compliant with git config syntax.

Given that is an unused feature, it might be best to remove it from the perspective of datalad-core. From the perspective of an extension wanting a dedicated logging hierarchy `name` should stay configurable, but should not impact the names of config switches.

**Comment by adswa** [comment]
I used your patch to check what tests fail, and I can't make sense of the failures I'm seeing:

```
FAILED datalad/tests/test_log.py::test_logging_to_a_file - 
AssertionError: There should be no color formatting in log files. Got: [ERROR  ] Oh my god, they killed Kenny 
FAILED datalad/tests/test_log.py::test_filters - 
AssertionError: assert 'log3' not in '[DEBUG] log...INFO] log3\n'
FAILED datalad/tests/test_log.py::test_log_progress_noninteractive_filter - 
AssertionError: assert 'Start' in ''
```

I've specifically looked into the first failure, the fact that logging to a file unexpectedly contains color, and I don't get why. 

This is what the test does:

```py
lgr = LoggerHelper("dataladtest-1").get_initialized_logger(logtarget=dst)
ok_(exists(dst))  # nothing was logged -- no file created

msg = "Oh my god, they killed Kenny"
lgr.error(msg)
with open(dst) as f:
    lines = f.readlines()
assert_equal(len(lines), 1, "Read more than a single log line: %s" %  lines)
line = lines[0]
ok_(msg in line)
ok_('\033[' not in line,
    msg="There should be no color formatting in log files. Got: %s" % line)
```

I can make the test fail with this minimal diff:

```diff
diff --git a/datalad/log.py b/datalad/log.py
index 646b9cdeb..79343f244 100644
--- a/datalad/log.py
+++ b/datalad/log.py
@@ -567,14 +567,14 @@ class LoggerHelper(object):
 
         Parameters
         ----------
         name :
         logtarget : string, optional
           If we want to use our logger for other log targets, while having
           a uniform control over them
         """
         self.name = name
         self.logtarget = logtarget
-        self.lgr = logging.getLogger(logtarget if logtarget is not None else name)
+        self.lgr = logging.getLogger(logtarget if logtarget is not None else 'dataladtest-1')
 ```
 
 where ``dataladtest-1`` is the exact content of the `name` variable.

**Comment by bpoldrack** [comment]
Dropping a note on a recollection, that may help figuring this out:

I think `LoggerHelper` somewhere does dynamically build config names to look up, where the logger's name is used to determine the first part. So, in your case, @adswa, it would start looking for some `dataladtest-1.log.XXX`.
That may mess with the detection of what formatting/colorization to use with the logger.

Yes, that's bad. An the intention of that code is long outdated and its anticipation of what may be useful never came to reality.

**Comment by adswa** [comment]
I tried grepping for places where this would happen, but must have missed them so far. Any pointer where that may be?

**Comment by bpoldrack** [comment]
@adswa: https://github.com/datalad/datalad/blob/maint/datalad/log.py#L581

**Comment by adswa** [comment]
Yes, that I have seen, too, and I've also done exactly what you proposed as well to check if it plays a part - it doesn't seem so.

**Comment by bpoldrack** [comment]
Ok, I probably spend too much time on this, but I think I figured at least one aspect of it.

This test (only looking at the logging to file one for now) *requires* that we have the logger created within in a different name space than the instance `datalad.lgr`.

Both, @mih's patch and your way of reproducing it, @adswa took that away. The result is, that the logger in the test inherits additional handlers from `datalad.lgr` (which it did not, when toplevel `'datalad'` logger was unrelated to this `'dataladtest-1'` logger. I am not quite sure how the colored `StreamHandler` for stderr screws up the `RotatingFileHandler` for `dst`, but somewhere in there is the issue, I think.

Implication not yet clear to me. Those handlers shouldn't really interfere with eachother. But I can see, that the created `RotatingFileHandler` and its Formatter are just fine on creation (not using coloring). If we end up with the color sequence in that file - interference is happening.
